src/Controller/ProductsController.php

```diff
@@ -35,4 +36,23 @@ public function apiProducts(SerializerInterface $serializer): Response
         $jsonContent = $serializer->serialize($products, 'json');
         return JsonResponse::fromJsonString($jsonContent);
     }
+
+    #[Route('/api/products/save', name: 'api_products_save', methods: ['POST'])]
+    public function apiProductsSave(Request $request): Response
+    {
+        $product_data = json_decode($request->getContent(), true);
+
+        $fields = ['productId', 'name', 'totalSOld'];
+
+        foreach ($fields as $field) {
```

Comment:
šis validavimo būdas veikia, tačiau norint sudėtingesnės validacijos reikėtų rašyti daug perteklinio kodo.
Rekomenduoju pasiskaityti/pabandyti Symfony Validator komponentą: https://symfony.com/doc/current/validation.html
jis leidžia atlikti sudėtingas validacijas visai paprastai, net nerašant daug kodo 🙂